zerver/tests/test_user_groups.py

```diff
@@ -1595,7 +1604,7 @@ def test_query_counts(self) -> None:
 
         with (
             mock.patch("zerver.views.user_groups.notify_for_user_group_subscription_changes"),
-            self.assert_database_query_count(11),
+            self.assert_database_query_count(15),
```

Comment:
When we were using the edit_policy setting, the default for who can edit all groups was member in the test and `hamlet` being a member of the group being tested, we never reached the `user_has_permission_for_group_setting`. After that, the intermediate condition for checking the creator was doing an early return until the last commit. 

The increase in the queries is because of us reaching the `user_has_permission_for_group_setting` for `has_user_group_access` this time around.